src/python/pants/backend/docker/goals/package_image.py

```diff
@@ -377,17 +391,41 @@ def get_build_options(
     if global_build_no_cache_option:
         yield "--no-cache"
 
+    if push_image:
+        if use_buildx_option is not True:
+            raise DockerImageOptionValueError(
+                f"The `--push` flag was set on the command line, but buildx is not enabled. "
+                f"Buildx must be enabled via the Docker subsystem options in order to use this flag."
+            )
+        yield "--push"
+
+
+@dataclass(frozen=True)
+class DockerBuildSetupRequest:
+    field_set: DockerPackageFieldSet
+    ignore_platforms: bool
+    push_image: bool = False
+
+
+@dataclass(frozen=True)
+class DockerBuildSetup:
+    process: Process
+    context: DockerBuildContext
+    context_root: str
+    tags: tuple[str, ...]
+    image_refs: tuple[ImageRefRegistry, ...]
+
 
 @rule
-async def build_docker_image(
```

Comment:
`build_docker_image` was split up so I can re-use the majority of it from publish